### PR TITLE
if no sectors need to recovery, checkNextRecoveries will return

### PR DESCRIPTION
**Pull request by Leozhang404**
after checkSectors(), there is no sectors to recovery, it should return directly

### CODE REVIEW

**Comment by Leozhang404** [comment]
please check it @magik6k @whyrusleeping

File: storage/wdpost_run.go [hunk 1]
```diff
@@ -150,6 +150,15 @@ func (s *WindowPoStScheduler) checkNextRecoveries(ctx context.Context, deadline
 	if err != nil {
 		return xerrors.Errorf("checking unrecovered sectors: %w", err)
 	}
+	
+	// check some unrecovered sectors can not recovered (such as cache file is missing)
```
Comment (magik6k) [inline storage/wdpost_run.go:154]:
```suggestion
	// if all sectors failed to recover, don't declare recoveries
```

File: storage/wdpost_run.go [hunk 2]
```diff
@@ -150,6 +150,15 @@ func (s *WindowPoStScheduler) checkNextRecoveries(ctx context.Context, deadline
...
 	}
+	
+	// check some unrecovered sectors can not recovered (such as cache file is missing) 
+	sbfCount, err := sbf.Count()
+	if err != nil {
+		log.Errorf(" sbf.Count() | %v", err)
+	} else if sbfCount == 0 {
+	 	log.Warnf("unrecovered exists sbf.Count() sectors can not recovery, exit this function.", err)
+	 return nil
+	}
```
Comment (magik6k) [inline storage/wdpost_run.go:161]: `Count` failing here should just return an error - if it fails, sending the message will fail anyways

```suggestion
		return xerrors.Errorf("counting recovered sectors: %w", err)
	}
	
	if sbfCount == 0 {
	 	log.Warnw("No recoveries to declare", "deadline", deadline, "faulty", uc)
	 	return nil
	}
```

**Comment by Leozhang404** [comment]
thanks @magik6k , you are right, please check and merge it.
(if all sectors failed to recover, don't declare recoveries)

**Pull request merged**